providers/Bandcamp/mod.ts

```diff
@@ -51,11 +51,11 @@ export default class BandcampProvider extends MetadataProvider {
 		const albumResult = this.supportedUrls.exec(url);
 		if (albumResult) {
 			const artist = albumResult.hostname.groups.artist!;
-			const { type, album } = albumResult.pathname.groups;
-			if (type && album) {
+			const { type, title } = albumResult.pathname.groups;
+			if (type && title) {
 				return {
 					type,
-					id: [artist, album].join('/'),
+					id: [artist, title].join('/'),
```

Comment:
I changed it to encode the type in the ID instead, but default to `album` if no type is specified in the ID, which I think should work. (BTW, I couldn't figure out where/how such URLs are generated.)

Reply:
The permalinks are generated here:
https://github.com/kellnerd/harmony/blob/f5eb66c39d6322e49a0818fb6106d9ed901ed840/server/state.ts#L20-L23
Since the Bandcamp provider IDs with their text blurb are quite long, what do you think about omitting the type `album` in `extractEntityFromUrl`?
That is defaulting to the old format `artist/title` for albums but using the new format `artist/type/title` only for tracks.

Reply:
> The permalinks are generated here:

I saw that, but where are the permalinks exposed? 😕 I didn't find them on the `/release` page or in the seeded edit note.

> Since the Bandcamp provider IDs with their text blurb are quite long, what do you think about omitting the type `album` in `extractEntityFromUrl`?
> That is defaulting to the old format `artist/title` for albums but using the new format `artist/type/title` only for tracks.

Sounds good to me!

Reply:
Permalinks are in the [edit note](https://github.com/kellnerd/harmony/blob/f5eb66c39d6322e49a0818fb6106d9ed901ed840/musicbrainz/seeding.ts#L131), but [only on production servers](https://github.com/kellnerd/harmony/blob/f5eb66c39d6322e49a0818fb6106d9ed901ed840/server/routes/release.tsx#L22).

My thought was not to seed edit notes with localhost URLs, but the different behavior for development servers is just not helpful in retrospect. You could also seed localhost URLs with a local production server ( `deno task server`), not only with `deno task dev`.

I think I will remove this special casing, I have already missed the possibility to review permalinks with a dev server myself a few times.
And the release lookup page itself should probably also expose the permalink.

Reply:
Ah okay, thank you for the explanation.

Maybe you can rewrite the domain in the seeded URLs to `harmony.pulsewidth.org.uk` if the target MusicBrainz server is `*.musicbrainz.org`? (Not sure if it's even possible to configure submission to a local MB server.)

Reply:
> Maybe you can rewrite the domain in the seeded URLs

It is not recommended to do that, as the server won't have the necessary snapshots cached for permalinks it hasn't generated itself.
If there is no snapshot for a given timestamp, a Harmony server will currently fallback to querying all providers again and use that data which is not cached and might have changed in the mean time.

> (Not sure if it's even possible to configure submission to a local MB server.)

https://github.com/kellnerd/harmony/blob/f5eb66c39d6322e49a0818fb6106d9ed901ed840/server/config.ts#L7-L8

It is possible using an environment variable.
And I just realized that this URL is not used for API requests as advertised 😅
But it might be better to have two separate settings for these two purposes anyway.

Reply:
Right, I just realized the permalinks are linked to a snapshot with a timestamp in the URL. Thanks for your patience in explaining things!

> It is possible using an environment variable.

Yep, that works perfectly (except for the API queries, now I want my money back!).